src/core/geometry/qgslinestring.cpp

```diff
@@ -443,6 +443,21 @@ double QgsLineString::length() const
   return length;
 }
 
+double QgsLineString::length3D() const
+{
+  double length = 0;
```

Comment:
If someone calls this on a LineString with no z, it'll crash. Can you add a check at the start here and if there's no z present, just return length() instead?

Reply:
Added.